### PR TITLE
Fix DefaultCachedClasspathTransformer not being an immutable cache

### PR DESCRIPTION
**Pull request by lptr**
It should be recognized among the other well-known immutable locations.

### CODE REVIEW

File: subprojects/core/src/main/java/org/gradle/internal/file/JarCache.java [hunk 1]
```diff
@@ -34,15 +33,10 @@ public JarCache(FileHasher fileHasher) {
 
     /**
      * Returns a cached copy of the given file. The cached copy is guaranteed to not be modified or removed.
-     *
-     * @param original The source file.
-     * @param baseDirFactory A factory that can provide a base directory for the file cache.
-     * @return The cached file.
      */
-    public File getCachedJar(File original, Factory<File> baseDirFactory) {
```
Comment (wolfs) [inline subprojects/core/src/main/java/org/gradle/internal/file/JarCache.java]: Any idea why a factory has been used in the first place?

**Comment by wolfs** [comment]
I see, those tests need to be adjusted since the constant is now somewhere else: https://github.com/gradle/gradle/blob/612c5d344c684d7e413433ec8aaa2a669c06f9f3/subprojects/core/src/integTest/groovy/org/gradle/initialization/buildsrc/BuildScriptClasspathIntegrationSpec.groovy#L266

File: subprojects/core/src/main/java/org/gradle/internal/file/JarCache.java (hunk 1 above)
Comment (lptr) [reply subprojects/core/src/main/java/org/gradle/internal/file/JarCache.java]: Nope. Maybe at some point it was expected that the cache dir would change, and was kept lazy?

**Comment by lptr** [comment]
OK, I moved some some stuff around, should be good now. Will merge if build succeeds.

**Pull request merged**